File: src/Java/InsertionSort.java
package Java;

public class InsertionSort {
    public static void insertionSort(int array[]) {
        int n = array.length;
        for (int j = 1; j < n; j++) {
            int key = array[j];
            int i = j-1;
            while ( (i > -1) && ( array [i] > key ) ) {
                array [i+1] = array [i];
                i--;
            }
            array[i+1] = key;
        }
    }

    public static void main(String a[]){
        int[] arr1 = {64, 34, 25, 12, 22, 11, 90,88};
        System.out.println("Before Insertion Sort");
        for(int i:arr1){
            System.out.print(i+" ");
        }
        System.out.println();

        insertionSort(arr1);

        System.out.println("After Insertion Sort");
        for(int i:arr1){
            System.out.print(i+" ");
        }
    }
}

File: src/Kotlin/InsertionSort.kt
package Kotlin

object InsertionSort {
    fun insertionSort(array: IntArray) {
        val n = array.size
        for (j in 1 until n) {
            val key = array[j]
            var i = j - 1
            while (i > -1 && array[i] > key) {
                array[i + 1] = array[i]
                i--
            }
            array[i + 1] = key
        }
    }

    @JvmStatic
    fun main(a: Array<String>) {
        val arr1 = intArrayOf(64, 34, 25, 12, 22, 11, 90, 88)
        println("Before Insertion Sort")
        for (i in arr1) {
            print("$i ")
        }
        println()
        insertionSort(arr1)
        println("After Insertion Sort")
        for (i in arr1) {
            print("$i ")
        }
    }
}
